src/main/java/frc/robot/Arm.java

```diff
@@ -0,0 +1,101 @@
+package frc.robot;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.REVPhysicsSim;
+import com.revrobotics.CANSparkLowLevel.MotorType;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class Arm extends SubsystemBase {
+    private double targetAngle;
+    private Encoder armEncoder;
+
+    private CANSparkMax motorOne;
+    private CANSparkMax motorTwo;
+    private PIDController pid;
+    private DigitalInput homeSwitch;
+
+    public Arm() {
+        homeSwitch = new DigitalInput(Constants.Arm.DIGITAL_INPUT_CHANNEL);
+        pid = new PIDController(Constants.Arm.P_COEFFICIENT, Constants.Arm.I_COEFFICIENT, Constants.Arm.D_COEFFICIENT);
+        armEncoder = new Encoder(Constants.Arm.ENCODER_CHANNEL_A, Constants.Arm.ENCODER_CHANNEL_B);
+        motorOne = new CANSparkMax(Constants.Arm.ARM_MOTOR_ONE, MotorType.kBrushless);
+        motorTwo = new CANSparkMax(Constants.Arm.ARM_MOTOR_TWO, MotorType.kBrushless);
+        if (Robot.isSimulation()) {
+            REVPhysicsSim.getInstance().addSparkMax(motorOne, DCMotor.getNEO(1));
+            REVPhysicsSim.getInstance().addSparkMax(motorTwo, DCMotor.getNEO(1));
+        }
+    }
+
+    public double getAngle() {
+        return ((double) armEncoder.get() / Constants.Arm.ENCODER_COUNT) * 360;
+    }
+
+    public boolean isOnTarget() {
+        return ((targetAngle - Constants.Arm.TARGET_RANGE_DEGREES <= getAngle())
+                && (targetAngle + Constants.Arm.TARGET_RANGE_DEGREES >= getAngle()));
+    }
+
+    public double getTarget() {
+        return targetAngle;
+    }
+
+    public void setTarget(double inputTarget) {
+        targetAngle = inputTarget;
+    }
+
+    public boolean isAtHomePostion() {
+        if (homeSwitch.get()) {
+            return true;
+        } else {
+            return (
+                (Constants.Arm.HOME_POSITION - Constants.Arm.HOME_POSITION_RANGE_DEGREES <= getAngle())
+                && 
+                (Constants.Arm.HOME_POSITION + Constants.Arm.HOME_POSITION_RANGE_DEGREES >= getAngle())
+                );
+
+        }
+    }
+
+    @Override
+    public void periodic() {
+        double motorVoltage = pid.calculate(targetAngle - getAngle());
+
+        if(homeSwitch.get()) {
+            //Check if encoder and home switch disagree
+            if(getAngle() > Constants.Arm.HOME_SWITCH_FAILSAFE_DEGREES) {
+                //Assume a sensor is broken, do not move arm.
+                motorOne.setVoltage(0);
+                motorTwo.setVoltage(0);
```

Comment:
I think it would be cool if you guys wrote a boolean method that let us know if the arm was "disabled" as judged by this code. No need to use it in a command yet, just make the method.